extensions-core/mysql-metadata-storage/src/main/java/org/apache/druid/metadata/storage/mysql/MySQLConnector.java

```diff
@@ -143,6 +144,7 @@ public MySQLConnector(
     datasource.setConnectionInitSqls(ImmutableList.of("SET sql_mode='ANSI_QUOTES'"));
 
     this.dbi = new DBI(datasource);
+    this.dbi.setSQLLog(new SQLProfileLogger());
```

Comment:
don't use `new` inside a constructor. This makes it difficult to test. Guice inject the logger instead. Should the logger be a singleton?